regression/contracts/quantifiers-exists-ensures-02/test.desc

```diff
@@ -1,19 +1,14 @@
 CORE
 main.c
---enforce-all-contracts
-^EXIT=10$
+--replace-all-calls-with-contracts
+^EXIT=0$
 ^SIGNAL=0$
-^VERIFICATION FAILED$
+^\[main.assertion.1\] line .* assertion len > 0 ==> found_zero: SUCCESS$
+^VERIFICATION SUCCESSFUL$
 --
+^warning: ignoring
 --
 The purpose of this test is to ensure that we can safely use __CPROVER_exists
-in __CPROVER_ensures clauses. By using the --enforce-all-contracts flag, 
-goto-instrument will transform the __CPROVER_ensures clauses into an
-assertion and the verification remains sound when using __CPROVER_exists.
+within negative contexts (replaced ENSURES clauses).
 
-Known Bug:
-We expect verification to fail because arr[i] is always equal to 0 for
-[0 <= i < 10]. In fact, we expect the (0 <= i && i < 10) statement to act as a
-range for i. However, in the current implementation of quantifier handling,
-the (0 <= i && i < 10) statement is not interpreted as an explicit range, but
-instead, as part of a logic formula, which causes verification to succeed.
+This is fully supported (without requiring full unrolling) with the SAT backend.
```

Comment:
What do you mean by `without requiring full unrolling`?

Reply:
May be there is a better way to phrase this. I was being lazy.

I meant to say that `exists x . (L <= x <= U) && P(x)` is not unrolled to `P(L) \/ P(L+1) \/ ... \/ P(U-1) \/ P(U)` in a negative context.

In a negative context symbolic or even no explicit bounds at all would work for `exists`, e.g. `assume(exists x . P(x))` works fine because CBMC just introduces a temporary variable `x` and then `assume(P(x))`.

Reply:
https://github.com/diffblue/cbmc/issues/6149